src/uct/tcp/tcp_sockcm_ep.h

```diff
@@ -0,0 +1,57 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2019.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#include "tcp_listener.h"
+
+
+typedef enum uct_tcp_sockcm_ep_state {
+    UCT_TCP_SOCKCM_EP_ON_SERVER      = UCS_BIT(0),
+    UCT_TCP_SOCKCM_EP_ON_CLIENT      = UCS_BIT(1),
+    UCT_TCP_SOCKCM_EP_INIT           = UCS_BIT(2),
+    UCT_TCP_SOCKCM_EP_CONNECTED      = UCS_BIT(3), /* connect()/accept()
+                                                      completed successfully */
+    UCT_TCP_SOCKCM_EP_SENDING        = UCS_BIT(4),
+    UCT_TCP_SOCKCM_EP_DATA_SENT      = UCS_BIT(5),
+    UCT_TCP_SOCKCM_EP_DATA_RECEIVING = UCS_BIT(6),
+    UCT_TCP_SOCKCM_EP_DATA_RECVED    = UCS_BIT(7),
+    UCT_TCP_SOCKCM_EP_DISCONNECTING  = UCS_BIT(8),  /* uct_ep_disconnect was
+                                                       called on the ep.
+                                                       this ep is not
+                                                       necessarily disconnected
+                                                       yet */
+    UCT_TCP_SOCKCM_EP_FAILED         = UCS_BIT(9)
+} uct_tcp_sockcm_ep_state_t;
+
+
+/**
+ * TCP sockcm endpoint communication context
+ */
+typedef struct uct_tcp_sockcm_ep_ctx {
+    void                        *buf;           /* Data buffer to send */
+    size_t                      total_length;   /* How much data to send */
+    size_t                      offset;         /* Next offset to send */
+} uct_tcp_sockcm_ep_ctx_t;
+
+
+/**
+ * TCP SOCKCM endpoint that is opened on a connection manager
+ */
+struct uct_tcp_sockcm_ep {
```

Comment:
typedef?

Reply:
in tcp_sockcm.h

Reply:
why dont define the structure in tcp_sockcm.h as well?

Reply:
since it's the ep, i think it should be in the ep header file..

Reply:
completely agree, but why not in the single place?
```
typedef struct uct_tcp_sockcm_ep {
...
} uct_tcp_sockcm_ep_t;
```